source/src/AppEventLoop.cc

```diff
@@ -48,84 +48,72 @@ namespace dqm4hep {
     
     //-------------------------------------------------------------------------------------------------
     
-    void AppEventLoop::postEvent(AppEvent *pAppEvent)
-    {
-      if(nullptr == pAppEvent)
+    void AppEventLoop::postEvent(AppEvent *pAppEvent) {
+      if(nullptr == pAppEvent) {
         return;
-      std::lock_guard<std::recursive_mutex> lock(m_queueMutex);      
+      }
       // push event in the queue
-      m_eventQueue.push_front(std::shared_ptr<AppEvent>(pAppEvent));      
-      // sort the event queue by priority
-      std::sort(m_eventQueue.begin(), m_eventQueue.end(), [](AppEventPtr lhs, AppEventPtr rhs){
-        return lhs->priority() < rhs->priority();
-      });
+      std::lock_guard<std::recursive_mutex> lock(m_queueMutex);      
+      m_eventQueue.insert(pAppEvent);   
     }
     
     //-------------------------------------------------------------------------------------------------
     
-    void AppEventLoop::sendEvent(AppEvent *pAppEvent)
-    {
-      if(nullptr == pAppEvent)
+    void AppEventLoop::sendEvent(AppEvent *pAppEvent) {
+      if(nullptr == pAppEvent) {
         return;
-        
-      // take ownership of event ptr
-      std::unique_ptr<AppEvent> ptr(pAppEvent);
-      
+      }
       // process an event
       this->processEvent(pAppEvent);
+      delete pAppEvent;
```

Comment:
I haven't followed the whole life cycle of pAppEvent but I didn't expect it to be consumed here.
Shouldn't it be deleted where it was created for more clarity? 
By where it was created I mean if it's created in a ctor I would naively expect it to be deleted in the dtor for example.

Reply:
By convention/definition the app event loop owns the app events. Always. So, as a user you create an event and post it, the ownership is transfered. It was also the case before

Reply:
Alright, it's fine. As you mentioned before it was handled by smart_ptr so this was transparent.